### PR TITLE
PLAT-310 call requests list

### PR DESCRIPTION
**Pull request by AndreyBronin**
- What I did
add requests list in execute shared state for deduplication incoming requests algorithm
unit test in progress
add check for duplicate request

- How I did it
add `SawRequests map[reference.Global]struct{}` to shared state

- How to verify it
unit test

### CODE REVIEW

File: ledger-core/v2/virtual/object/object.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ type Info struct {
 
 	AwaitPendingOrdered smachine.BargeIn
 
+	SawRequests map[reference.Global]struct{}
```
Comment (sexyhamster) [inline ledger-core/v2/virtual/object/object.go:49]: SeenRequests/KnownRequests. Not sure about the map, we have to not only be checking if requests is in list, but also we will need to find EarliestKnownPulse. Maybe OrderedMap would suit here better (it should already be in our codebase)

Comment (sexyhamster) [reply ledger-core/v2/virtual/object/object.go:49]: but maybe just earliest/latest request field would do the job

Comment (AndreyBronin) [reply ledger-core/v2/virtual/object/object.go:49]: @ruz do we need EarliestKnownPulse or ordered list for deduplication algorithm?

Comment (ruz) [reply ledger-core/v2/virtual/object/object.go]: guys, if you look at the task and the story then you can find answer to these questions.

**Pull request merged**